File: app/src/main/java/com/tstudioz/iksica/MyApplication.java
package com.tstudioz.iksica;

import android.app.Application;
import android.content.Context;

import androidx.multidex.MultiDex;

import com.google.android.gms.ads.MobileAds;
import com.orhanobut.hawk.Hawk;
import com.tstudioz.iksica.Data.DatabaseHelper;
import com.tstudioz.iksica.Utils.NetworkMonitor;
import com.tstudioz.iksica.Utils.OkHttpClientX;


import io.realm.Realm;
import timber.log.Timber;

public class MyApplication extends Application {

    @Override
    public void onCreate() {

        if (BuildConfig.DEBUG) {
            Timber.plant(new Timber.DebugTree());
        }

        super.onCreate();

        NetworkMonitor.createInstance(getApplicationContext());
        OkHttpClientX.createInstance(getApplicationContext());

        Realm.init(this);
        Hawk.init(getApplicationContext()).build();
        DatabaseHelper.Companion.createInstance(getApplicationContext());

        // MobileAds.initialize(this, "ca-app-pub-5944203368510130~6211911487");

    }

    @Override
    protected void attachBaseContext(Context base) {
        super.attachBaseContext(base);
        MultiDex.install(this);
    }

}

File: app/src/main/java/com/tstudioz/iksica/MyApplication.kt
package com.tstudioz.iksica

import android.app.Application
import android.content.Context
import androidx.multidex.MultiDex
import com.google.android.gms.ads.MobileAds
import com.orhanobut.hawk.Hawk
import com.tstudioz.iksica.DI.defaultModule
import com.tstudioz.iksica.DI.viewModelModule
import org.koin.android.ext.koin.androidContext
import org.koin.android.ext.koin.androidLogger
import org.koin.core.context.startKoin
import timber.log.Timber
import timber.log.Timber.DebugTree

class MyApplication : Application() {
    override fun onCreate() {
        super.onCreate()

        if (BuildConfig.DEBUG) {
            Timber.plant(DebugTree())
        }

        startKoin {
            androidLogger()
            androidContext(this@MyApplication)
            modules(listOf(defaultModule, viewModelModule))
        }

        Hawk.init(applicationContext).build()
        MobileAds.initialize(this)
    }

    override fun attachBaseContext(base: Context) {
        super.attachBaseContext(base)
        MultiDex.install(this)
    }
}